ARCHITECTURE.md

```diff
@@ -136,8 +261,224 @@ Mutation webhooks meant for intercepting user Kubernetes Resources, so they can
+
+### `pkg/util`
+
+Utility packages that are used throughout the codebase. Packages contain Kubernetes resources as well as constants that are used frequently for example.
+
+### `pkg/version`
+
+Package that deals with semantic versions. Is able to extract said versions, compare them and so on.
+
+### `pkg/webhook`
+
+Main logic of the webhook. Both `MutationWebhook` as well as `ValidationWebhook` are handled here.
+
+### `pkg/webhook/mutation/namespace`
+
+Main logic of the `MutationWebhook` mutating `Namespaces`. Does the mapping between the `Namespaces` and `DynaKube` from the `Namespace's` side.
+
+### `pkg/webhook/mutation/pod`
+
+Main logic of the `MutationWebhook` mutating `Pods`. Does the mapping between the `Namespaces` and `DynaKube` from the `Namespace's` side.
+
+### `pkg/webhook/mutation/pod/metadata`
+
+Package that deals with necessary metadata for `Pods`. Sets up and prepares `Containers`, `Envs` and `Volumes` that are needed for mutating `Pods`.
+
+### `pkg/webhook/mutation/pod/oneagent`
+
+Package that deals with necessary metadata for `OneAgent` related metadata. Sets up for example `Volumes` which hold necessary files that are needed for `OneAgent` installation.
+
+### `pkg/webhook/util`
+
+Utils package that does operations like checking `EnvVars` in `Containers`.
+
+### `pkg/webhook/validation`
+
+Main logic for the `ValidationWebhook`. Deals with both `DynaKube` and `EdgeConnect`. Checks if the configuration of said Resources is valid and can be applied. Gives warnings or even errors if there is some misconfiguration.
+
+### `test`
+
+E2E testing code. Unit tests are NOT found here, they are in the same module that they are testing, as that is the Golang convention.
+
+### `test/features`
+
+Respective feature for each scenario. Those features are then called within the tests and ran through.
+
+### `test/helpers`
+
+Package that has helpers which are needed for the E2E tests.
+
+### `test/mocks`
+
+Package that has mocks which are needed for the E2E tests.
+
+### `test/project`
+
+Package that sets up the root directory that will be used within the E2E tests.
+
+### `test/scenarios`
+
+Package that has all scenarios that are tested by the E2E tests. Those scenarios have a `Test*` function that calls the features from the `feature` package and runs the tests.
+
+### `test/testdata`
+
+Has every data that is needed to run the tests. This reaches from `Deployments` up to `Secrets` to be able to install the `Operator`.
+
+---
+Find here some very basic visualizations of the Operator with its components:
+
+## Overall functionality
+
+```mermaid
+graph 
+    DK[fa:fa-dk Dynakube] --> WH(fa:fa-file Dynatrace-Webhook):::wh
+    DK[fa:fa-dk Dynakube] --> OP(fa:fa-op Dynatrace-Operator):::op
+    DK[fa:fa-dk Dynakube] --> CSI(fa:fa-csi CSI Driver):::csi
+    AG --> Dynatrace-Cluster
+    subgraph K8S Node
+    WH
+    OP -->|deploys| OA[OneAgent]:::oa
+    OP -->|deploys| AG[ActiveGate]:::ag
+    CSI --> P[Pods]:::pod
```

Comment:
what are the pods for?

Reply:
My idea was to sort of start with a very rough overview and if you guys think this is a good idea maybe go step by step more into detail, Of course I would also then describe what the Pods do.